railties/lib/rails/commands/statistics/statistics_command.rb

```diff
@@ -26,12 +26,21 @@
   %w(Channel\ tests     test/channels),
   %w(Integration\ tests test/integration),
   %w(System\ tests      test/system),
-].collect do |name, dir|
-  [ name, "#{File.dirname(Rake.application.rakefile_location)}/#{dir}" ]
-end.select { |name, dir| File.directory?(dir) }
+]
 
-desc "Report code statistics (KLOCs, etc) from the application or engine"
-task :stats do
-  require "rails/code_statistics"
-  CodeStatistics.new(*STATS_DIRECTORIES).to_s
+module Rails
+  module Command
+    class StatisticsCommand < Base # :nodoc:
+      desc :stats, "Report code statistics (KLOCs, etc) from the application or engine"
+      def perform
+        require "rails/code_statistics"
+
+        stat_directories = STATS_DIRECTORIES.collect do |name, dir|
+          [name, "#{File.dirname(Rake.application.rakefile_location)}/#{dir}"]
```

Comment:
I think we can now do the following:
```suggestion
          [name, Rails::Command.application_root.join(dir)]
```
Similar to https://github.com/rails/rails/blob/a51364dcf1dab34bc6cfc9aeca56a3b0ecf823ad/railties/lib/rails/commands/restart/restart_command.rb#L9